browser/javascript/v2/react-app/src/ampli/index.js

```diff
@@ -0,0 +1,590 @@
+/* tslint:disable */
+/* eslint-disable */
+/**
+ * Ampli - A strong typed wrapper for your Analytics
+ *
+ * This file is generated by Amplitude.
+ * To update run 'ampli pull browser-ts-ampli-v2'
+ *
+ * Required dependencies: @amplitude/analytics-browser@^0.5.0
+ * Tracking Plan Version: 0
+ * Build: 1.0.0
+ * Runtime: browser:javascript-ampli-v2
+ *
+ * [View Tracking Plan](https://data.amplitude.com/test-codegen/Test%20Codegen/events/main/latest)
+ *
+ * [Full Setup Instructions](https://data.amplitude.com/test-codegen/Test%20Codegen/implementation/browser-js-ampli-v2)
+ */
+
+import * as amplitude from '@amplitude/analytics-browser';
+
+/**
+ * @typedef BaseEvent
+ * @type {object}
+ * @property {string} event_type
+ * @property {Object.<string, *>} [event_properties]
+ */
+
+/**
+ * @typedef Plan
+ * @type {object}
+ * @property {string} [branch]
+ * @property {string} [source]
+ * @property {string} [version]
+ * @property {string} [versionId]
+ */
+
+/**
+ * @typedef LoadClientOptions
+ * @type {object}
+ * @property {string} [apiKey]
+ * @property {Object} [options]
+ * @property {AmplitudeClient} [instance]
+ */
+
+/**
+ * @typedef LoadOptions
+ * @type {object}
+ * @property {'development'|'production'} [environment]
+ * @property {boolean} [disabled]
+ * @property {LoadClientOptions} [client]
+ */
+
+/**
+ * @typedef {Object} EventOptions
+ * @type {object}
+ */
+
+/**
+ * @typedef {Object} IdentifyOptions
+ * @type {object}
+ */
+
+/**
+ * @typedef {Object} GroupOptions
+ * @type {object}
+ */
+
+/**
+ * @typedef ApiKey
+ * @type {object}
+ * @property {string} development
+ * @property {string} production
+ */
+export const ApiKey = {
+  development: '',
+  production: '',
+};
+
+/**
+ * @typedef {Object} EventTemplate
+ * @param {number} [optional_template_property] optional_template_property description
+ * @param {string} required_template_property required_template_property description
+ */
+
+/**
+ * @typedef {Object} SourceTemplate
+ * @param {'Value 1'|'Value 2'} [optionalEnum] description for context optionalEnum
+ * @param {string} requiredString description for context requiredString
+ */
+
+export const SpecialEventType = {
+  IDENTIFY: '$identify',
+  GROUP_IDENTIFY: '$groupidentify',
+};
```

Comment:
Is there a way to get this directly from core SDK instead? Or use event builders for Identify, Group events below?

Reply:
In the TS SDK, it is enum type, so we cannot get it from core SDK directly. The reason we need this is that this type is exposed to users and user can import the `SpecialEventType` to their code base. This part is aligned with https://github.com/amplitude/javascript/blob/6a06aff01d34934bb57282c0d74f3fea36900a3b/server/packages/stargate/codegen/javascript/ampli/v1/template-javascript-browser.ts#L90-L93